### PR TITLE
Add morse application

### PR DESCRIPTION
**Pull request by fgaz**
~~This is still missing a proper icon and screenshot~~

Signed-off-by: Francesco Gazzetta <fgaz@fgaz.me>

### CODE REVIEW

File: wasp/apps/morse.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+There is a preview of the next letter at the bottom of the screen.
+
+
+.. figure:: res/MorseApp.png
+    :width: 179
+
+.. note::
+
+    Due to it's niche purpose this app is not included by default in the flash
+    images. It must be separately installed and enabled.
```
Comment (fgaz) [inline wasp/apps/morse.py:21]: Oh oops, this is a leftover. As discussed in IRC, I'm including it in the default image. I'll delete the note when I add the images

**Comment by daniel-thompson** [comment]
You know a screenshot is just the 's' key in the simulator? Filename will be correct and in the right place so from there it is just a git add and you're done.
  

  

  

  
  
  
  
  
>

**Comment by fgaz** [comment]
Okay, I did that and some minor cleanup! This should be ready for review now

**Comment by fgaz** [comment]
I just noticed I was on an older micropython checkout (https://github.com/daniel-thompson/micropython/tree/048bb2da83d0cbe73a829e461bd4f22b86c282bc), updated the submodule and now the .text section of the firmware is 1kB too big :-/

I think I can shave off ~500 bytes by storing the lookup table in a different way, but I'd still be 500 bytes over the limit. Any tips? Am I doing anything particularly expensive? Or are there other apps that need optimization?

**Comment by daniel-thompson** [comment]
Also looks like the autotests are failing  (not enough FLASH space).

**Comment by fgaz** [comment]
I resolved the conversations and heavily optimized the letter lookup, but it's still 224 bytes over the limit. How should we proceed?

**Comment by daniel-thompson** [comment]
> I resolved the conversations and heavily optimized the letter lookup, but it's still 224 bytes over the limit. How should we proceed?

For now, app will need to be removed from manifest and software app list.

Does the app load using the new apps/ directory feature?

**Comment by fgaz** [comment]
It does. I'll move it there then

**Comment by fgaz** [comment]
I moved it, but now it doesn't get tested anymore. Do I have to expand the tests to `apps/` too?

**Comment by fgaz** [comment]
@daniel-thompson I expanded the tests

File: wasp/apps/morse.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
+# SPDX-License-Identifier: LGPL-3.0-or-later
+# Copyright (C) 2021 Francesco Gazzetta
+
+"""Morse translator and notepad
+~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (daniel-thompson) [inline wasp/apps/morse.py]: This line is triggering CI failures (the underlining should line up with the d in pad)

Comment (fgaz) [reply wasp/apps/morse.py]: Fixed!

**Comment by fgaz** [comment]
Great! This unblocks my other prs. I'll rebase them soon.

**Pull request merged**